probe_scraper/probe_expiry_alert.py

```diff
@@ -5,30 +5,32 @@
 from collections import defaultdict
 import argparse
 import datetime
-import logging
 import os
 import tempfile
 
 import requests
 
-from probe_scraper import emailer
 from probe_scraper.parsers.events import EventsParser
 from probe_scraper.parsers.histograms import HistogramsParser
 from probe_scraper.parsers.scalars import ScalarsParser
 from probe_scraper.parsers.utils import get_major_version
 
-FROM_EMAIL = "telemetry-alerts@mozilla.com"
-DEFAULT_TO_EMAIL = "dev-telemetry-alerts@lists.mozilla.org"
+BUGZILLA_BUG_URL = "https://bugzilla.mozilla.org/rest/bug"
+BUGZILLA_USER_URL = "https://bugzilla.mozilla.org/rest/user"
 
 BASE_URI = "https://hg.mozilla.org/mozilla-central/raw-file/tip/toolkit/components/telemetry/"
 HISTOGRAMS_FILE = "Histograms.json"
 SCALARS_FILE = "Scalars.yaml"
 EVENTS_FILE = "Events.yaml"
 
-EMAIL_BODY_FORMAT_STRING = """
-The following Firefox probes have either expired or will expire in the next major Firefox nightly release: version {next_version} [1].
+BUG_WHITEBOARD_TAG = "probe-expiry-alert"
+BUG_SUMMARY_TEMPLATE = "Remove or update probe expiring in Firefox {version}: {probe}"
+BUG_RESOLVED_COMMENT_TEMPLATE = "This probe is no longer expiring in Firefox {version}."
 
-{probes}
+BUG_DESCRIPTION_TEMPLATE = """
+The following Firefox probe will expire in the next major Firefox nightly release: version {version} [1].
```

Comment:
This sounds like a great suggestion. We could automate it if the prefix is up to an underscore.

Reply:
Note: This algorithm is a technical interview question to a _tee_.

Reply:
Consider that a gift to you : )

Reply:
Grouping by component and email set sounds good.  Probe names seems a little all over the place so I don't think prefixes will work consistently.  For example the last set of expiring probes had:
```
security.javascriptLoad#javascriptLoad
STARTUP_CACHE_REQUESTS
WEBEXT_USER_SCRIPT_INJECTION_MS_BY_ADDONID
TOUCHBAR_BUTTON_PRESSES
QM_REPOSITORIES_INITIALIZATION_TIME_V2
webrtc.sdp.parser_diff
startup.profile_selection_reason
script.preloader.mainthread_recompile
networking.ftp_opened_channels_listings
```

Reply:
Oh, we should just be able to group by email, right? What if we create a bug for each email instead of each probe.

Reply:
That's how email alerts are working now but if there are multiple emails in a set of related probes it would be better to just make one bug and needinfo all the emails.

Reply:
Okay. In that case let's make a one-off that finds all probes without a valid bugzilla email and we go and require that those teams add a valid bugzilla email. We can perhaps add a check as well that verifies this moving forward. What do you think @Ben-Wu @chutten?

Reply:
I checked all the probes active in nightly >= 76 and got this:
```
Total valid emails: 34 / 66
Probes with valid accounts by type:
scalar: 49 / 97
event: 0 / 22
histogram: 78 / 124
```

About half the probes have valid bugzilla accounts.  I'm worried that it would be too prohibitive to require a valid account.  Most of the invalid accounts are either email lists or email aliases.  I think that sending an email with the bug link to emails that don't have a bugzilla account is sufficient and definitely better than what we currently have.  So we're sending an email and creating a bug that will hopefully be triaged.  I'm sure @chutten will have more thoughts on this though

Reply:
Prefixes don't have to work consistently, it'll just add a little pressure for people to organize their histogram names : )

I think creating a bug
* per category/prefix (trying to keep the number of bugs filed down)
* in the component of the bug_number with the highest id (default to Firefox::General)
* with needinfos for the accounts there are
* and emails with links to the ones there aren't

is the optimal play for now. At the very least an unprioritized bug in a vaguely-relevant component will get the triage owners to triage it.